engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -45,119 +31,149 @@
 public class ChunkProcessingPipeline {
 
     private static final int NUM_TASK_THREADS = 4;
+    private static final int NUM_CHUNKS_AT_ONCE = 8;
     private static final Logger logger = LoggerFactory.getLogger(ChunkProcessingPipeline.class);
 
+    private final InitialChunkProvider initialChunkProvider;
+    private final Object waitForNewChunks = new Object();
+    private final Thread[] poolThreads = new Thread[NUM_TASK_THREADS];
```

Comment:
Just call this `threadPool`? (I think "thread pool" is quite a common term, and that's probably also what I'd be searching/looking for when debugging this class).

Reply:
It's so much a common term that I get _suspicious_ when I see us implementing our own like this.

Reply:
There used to also be a `ThreadPoolExecutor`, but now each worker thread fetches its own work. We actually don't need to store the threads in an array like this anymore, now that it doesn't interrupt them to stop them.